crates/bevy_pbr/src/material.rs

```diff
@@ -248,6 +254,91 @@ pub trait Material: Asset + AsBindGroup + Clone + Sized {
     }
 }
 
+#[derive(Default)]
+pub struct MaterialsPlugin {
+    /// Debugging flags that can optionally be set when constructing the renderer.
+    pub debug_flags: RenderDebugFlags,
+}
+
+impl Plugin for MaterialsPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins((PrepassPipelinePlugin, PrepassPlugin::new(self.debug_flags)));
+        if let Some(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app
+                .init_resource::<EntitySpecializationTicks>()
+                .init_resource::<SpecializedMaterialPipelineCache>()
+                .init_resource::<SpecializedMeshPipelines<MaterialPipelineSpecializer>>()
+                .init_resource::<LightKeyCache>()
+                .init_resource::<LightSpecializationTicks>()
+                .init_resource::<SpecializedShadowMaterialPipelineCache>()
+                .add_systems(
+                    Render,
+                    (
+                        specialize_material_meshes
+                            .in_set(RenderSystems::PrepareMeshes)
+                            .after(prepare_assets::<RenderMesh>)
+                            .after(collect_meshes_for_gpu_building)
+                            .after(set_mesh_motion_vector_flags),
+                        queue_material_meshes.in_set(RenderSystems::QueueMeshes),
+                    ),
+                )
+                .add_systems(
+                    Render,
+                    (
+                        prepare_material_bind_groups,
+                        write_material_bind_group_buffers,
+                    )
+                        .chain()
+                        .in_set(RenderSystems::PrepareBindGroups),
+                )
+                .add_systems(
+                    Render,
+                    (
+                        check_views_lights_need_specialization.in_set(RenderSystems::PrepareAssets),
+                        // specialize_shadows also needs to run after prepare_assets::<PreparedMaterial>,
+                        // which is fine since ManageViews is after PrepareAssets
+                        specialize_shadows
+                            .in_set(RenderSystems::ManageViews)
+                            .after(prepare_lights),
+                        queue_shadows.in_set(RenderSystems::QueueMeshes),
+                    ),
+                );
+
+            #[cfg(feature = "meshlet")]
+            render_app.add_systems(
```

Comment:
Now that these don't depend on material generics, lets move them into the meshlet plugin.